app/controllers/places_controller.rb

```diff
@@ -31,13 +31,36 @@ def add
     
         redirect_to search_path
     end
+
+    def destroy
+      id = params[:place][:id]
+      place_type = params[:place][:type]
+
+      current_user.public_send(place_type).delete(id)
```

Comment:
It can be dangerous to let the user delete any record that belongs to them. Without polymorphic associations, you can ensure that the type can only be one of `["attractions", "hotels", "restaurants"]` before proceeding, though that's a bit messy.